.github/workflows/ecr-shared-promote-prod.yml

```diff
@@ -23,13 +23,16 @@ on:
         required: false
         type: string
 
+permissions: read-all
+
 # Set defaults
 defaults:
   run:
     shell: bash
 
 jobs:
   deploy:
+    if: ${{ github.ref == 'refs/heads/main' || github.event.release.target_commitish == 'main' }}
     name: Promote Build to Prod
     # Download from Stage then upload to Prod
```

Comment:
Unrelated to this fix... we just always deploy the latest tag in stage when releasing to prod? Can stage be deployed any other way than by a merge to `main`? If so, we may want to rethink this as well (separate from this bug fix which I'll approve as it looks good).

Reply:
Hmmm... not 100% sure I understand your comment. 😁 

When "releasing to prod", we don't deploy/redeploy anything in stage. When "releasing to prod" we simply copy the latest release from stage and deploy that copy to prod, without changing anything in stage. The caller workflow is located in the app repo (here is the [geo-harvester prod-promote caller workflow](https://github.com/MITLibraries/geo-harvester/blob/1149ca71de134c4fc28af484ab9321cca6d208d4/.github/workflows/prod-promote.yml)) which has the following trigger:

```yaml
on:
  workflow_dispatch:
  release:
    types: [published]
```

The [stage-build caller workflow](https://github.com/MITLibraries/geo-harvester/blob/1149ca71de134c4fc28af484ab9321cca6d208d4/.github/workflows/stage-build.yml) has this trigger:

```yaml
on:
  workflow_dispatch:
  push:
    branches:
      - main
    paths-ignore:
      - '.github/**'
```

Which means that it's either a commit to `main` that triggers a deploy to Stage-Workloads (and `main` is a protected branch, so that commit should only ever come from merging a PR) or a `workflow_dispatch`. It's the `workflow_dispatch` that technically allows for deploying a feature branch to Stage-Workloads (since we don't do any branch chacking for manual runs for the stage workflows.

Does that help? or make things more complicated?

Reply:
Can this happen?

Developer 1: I have just merged my code and main was deployed to staging, I'll talk with stakeholders to have them check it before tagging for prod

Developer 2: <A few days later>. I'd like to see something in staging, let me deploy this sketchy feature branch to staging with workflow dispatch.

Developer 1: <Another few days later>. Okay, stakeholders approved, I'll tag main to deploy it to prod.

Developer 1: wtf? sketchy feature branch just got deployed to prod but I tagged main... why didn't it deploy main?

Reply:
So without the story mode, can only main be deployed to stage or can anything be deployed there?

Reply:
without running some tests in my testing repos, I'm 90%-99% sure that your story could happen. That is, a branch other than `main` could be deployed to Stage-Workloads (via the `workflow_dispatch` trigger in the caller workflow).

Reply:
Yeah me too. I think documenting this would be fine for now (i.e. before tagging a release it is expected to confirm that staging is in the correct state as we deploy the same container). I'm not sure how else to handle this other than to confirm the SHA in staging matches `main` before proceeding? Or just skipping the efficient process and just rebuilding the container (which I don't love for different reasons).

Reply:
I have confirmed in my testing repos that it is 100% possible that your story can happen. That is, it's possible with the current caller and shared workflows for a feature branch to get deployed after the `main` branch in Stage-Workloads so that a long pause between a PR merge to `main` (with a 2nd developer manually triggering a stage deploy from a feature branch) and a tagged release on `main` will result in the feature branch version of the container to get deployed to Prod-Workloads.

The good news is that we do tag the container in ECR with both `latest` and the commit SHA along with the date it was pushed to ECR:
![image](https://github.com/MITLibraries/.github/assets/60672715/42813336-be4a-44d1-bac8-ed61f91aec95)

So, for now, I think that documentation on the manual check by the developer issuing the tagged release will suffice. But, I'm not exactly sure where to document that...

Reply:
I'm going to move forward with merging this PR. The discussion here has been noted in [IN-992](https://mitlibraries.atlassian.net/browse/IN-992), an InfraEng/DataEng backlog ticket to update the documentation in our app repos .